examples/c/ex_col_store.c

```diff
@@ -0,0 +1,555 @@
+    start_time_cursor->set_key(start_time_cursor, start_time);
+    error_check(start_time_cursor->search_near(start_time_cursor, &exact));
+    switch (exact) {
+    case -1:
+        ret = start_time_cursor->next(start_time_cursor);
+        break;
+    default:
+        break;
+    }
+
+    if (ret == 0) {
+        error_check(session->join(session, join_cursor, start_time_cursor, "compare=ge"));
+    } else {
+        return false;
+    }
+
+    /* Find the ending record closest to desired end time. */
+    end_time_cursor->set_key(end_time_cursor, end_time);
+    error_check(end_time_cursor->search_near(end_time_cursor, &exact));
+    switch (exact) {
+    case 1:
+        ret = end_time_cursor->prev(end_time_cursor);
+        break;
+    default:
+        break;
+    }
+    if (ret == 0) {
+        error_check(session->join(session, join_cursor, end_time_cursor, "compare=le"));
+    } else {
+        return false;
+    }
+
+    /* Initialize maximum temperature to temperature of the first record. */
+    join_cursor->next(join_cursor);
+    if (ret != 0) {
+        return false;
+    }
+
+    error_check(join_cursor->get_key(join_cursor, &recno));
+    error_check(join_cursor->get_value(join_cursor, &hour, &temp));
+    *result = temp;
+
+    while ((ret = join_cursor->next(join_cursor)) == 0) {
+        error_check(join_cursor->get_key(join_cursor, &recno));
+        error_check(join_cursor->get_value(join_cursor, &hour, &temp));
+
+        if (temp > *result) {
+            *result = temp;
+        }
+
+        /* For debugging purposes */
+        printf("ID %" PRIu64, recno);
+        printf(": hour %" PRIu16 " temp: %" PRIu8 "\n", hour, temp);
+    }
+
+    return true;
+}
+
+/*! [Obtains the average data across all fields given a specific location] */
+void
+average_data(WT_SESSION *session)
+{
+    WT_CURSOR *loc_cursor;
+    const char *day;
+    const char *country;
+    uint64_t recno;
+    uint16_t hour, pressure, loc_long, loc_lat;
+    uint8_t temp, humidity, wind, feels_like_temp;
+    int ret, num_rec = 5, exact;
+    unsigned int count = 0;
+    unsigned int rec_arr[5] = {0, 0, 0, 0, 0};
+
+    /* Open a cursor to search for the location, currently RUS */
+    error_check(
+      session->open_cursor(session, "index:weathertable:country", NULL, NULL, &loc_cursor));
+    loc_cursor->set_key(loc_cursor, "RUS\0\0");
+    error_check(loc_cursor->search_near(loc_cursor, &exact));
+
+    /* Error handling in the case RUS is not found*/
+    if (exact != 0) {
+        printf("Location not found!");
+        return;
+    }
+
+    /* Populate the array with the totals of each of the columns */
+    while ((ret = loc_cursor->next(loc_cursor)) == 0) {
+        error_check(loc_cursor->get_key(loc_cursor, &recno));
+        error_check(loc_cursor->get_value(loc_cursor, &day, &hour, &temp, &humidity, &pressure,
+          &wind, &feels_like_temp, &loc_lat, &loc_long, &country));
+
+        if (!strcmp("RUS", country)) {
```

Comment:
This does not make sense, we are averaging for values that are not from RUS ?

Also as stated earlier, please do not promote result from strcmp into a boolean

`if (strcmp("RUS", country) != 0)`

But do you instead want to :

`if (strcmp("RUS", country) == 0)`